pkg/controller/controller.go

```diff
@@ -512,6 +514,42 @@ func unmarshalParameters(in []byte) (map[string]interface{}, error) {
 	return parameters, nil
 }
 
+// fetchSecretParameters requests and returns the contents of the given secret as a map
+func fetchSecretParameters(kubeClient kubernetes.Interface, namespace string, secretRef *v1.LocalObjectReference) (map[string]interface{}, error) {
+	if kubeClient == nil {
+		return nil, errors.New("kubeClient is not defined")
+	}
+	secret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]interface{})
```

Comment:
Sorry for piggy-backing this question on this PR, but I have a golang question... is there a fundamental difference between:
```
result := make(map[string]interface{})
```
and
```
result := map[string]interface{}{}
```
I don't think there is but I just wanted to check.